packages/ra-ui-materialui/src/list/FilterForm.js

```diff
@@ -69,53 +69,64 @@ const sanitizeRestProps = ({
     ...props
 }) => props;
 
-export class FilterForm extends Component {
-    componentDidMount() {
-        this.props.filters.forEach(filter => {
+const FilterForm = ({
+    filters,
+    displayedFilters,
+    initialValues,
+    hideFilter,
+    className,
+    classes,
+    resource,
+    ...rest
+}) => {
+    useEffect(() => {
+        filters.forEach(filter => {
             if (filter.props.alwaysOn && filter.props.defaultValue) {
                 throw new Error(
                     'Cannot use alwaysOn and defaultValue on a filter input. Please set the filterDefaultValues props on the <List> element instead.'
                 );
             }
         });
-    }
-
-    getShownFilters() {
-        const { filters, displayedFilters, initialValues } = this.props;
-
-        return filters.filter(
-            filterElement =>
-                filterElement.props.alwaysOn ||
-                displayedFilters[filterElement.props.source] ||
-                typeof lodashGet(initialValues, filterElement.props.source) !==
-                    'undefined'
-        );
-    }
-
-    handleHide = event =>
-        this.props.hideFilter(event.currentTarget.dataset.key);
+    }, [filters]);
+
+    const shownFilters = useMemo(
+        () =>
+            filters.filter(
+                filterElement =>
+                    filterElement.props.alwaysOn ||
+                    displayedFilters[filterElement.props.source] ||
+                    typeof lodashGet(
+                        initialValues,
+                        filterElement.props.source
+                    ) !== 'undefined'
+            ),
+        [displayedFilters, filters, initialValues]
```

Comment:
`filters` changes on every render, so the `useMemo` here is useless.

Reply:
The `useEffect` [here](https://github.com/marmelab/react-admin/blob/0282a34e3869838d0aebf9a40ca899bc41d991bf/packages/ra-ui-materialui/src/list/FilterForm.js#L82) would be useless too?

Reply:
yes, that `useEffect` should use an empty array as second argument